extensions/amp-sticky-ad/1.0/amp-sticky-ad.js

```diff
@@ -112,6 +114,41 @@ class AmpStickyAd extends AMP.BaseElement {
     return Promise.resolve();
   }
 
+  /** @override */
+  upgradeCallback() {
+    if (!isExperimentOn(this.win, 'amp-sticky-ad-to-amp-ad')) {
+      return null;
+    }
+
+    const children = realChildElements(this.element);
+    userAssert(
+      children.length == 1 && children[0].tagName == 'AMP-AD',
+      'amp-sticky-ad must have a single amp-ad child'
+    );
+
+    const ad = children[0];
+    const enableConversion = Math.random() < 0.5;
+
+    const adType = (ad.getAttribute('type') || '').toLowerCase();
+    if (adType == 'doubleclick' || adType == 'adsense') {
+      addExperimentIdToElement(enableConversion ? '31061856' : '31061855', ad);
+    }
+
+    if (!enableConversion) {
+      return null;
+    }
+
+    ad.setAttribute('sticky', 'bottom');
+    this.element.parentElement.replaceChild(ad, this.element);
+    return Services.extensionsFor(this.win)
+      .loadElementClass('amp-ad', '0.1')
+      .then((AmpAd) => {
+        const impl = new AmpAd(ad);
+        impl.upgradeCallback();
```

Comment:
Why are you calling `upgradeCallback` manually?

Reply:
I thought that an element cannot be upgraded the second time. Is that wrong?

Reply:
@jridgewell tested and it does upgrade again. Modified to not call upgrade manually.